internal/ethapi/api.go

```diff
@@ -51,21 +51,32 @@ const defaultGas = uint64(90000)
 // It offers only methods that operate on public data that is freely available to anyone.
 type PublicEthereumAPI struct {
 	b Backend
+	solcPath *string
+	solc     **compiler.Solidity
 }
 
 // NewPublicEthereumAPI creates a new Etheruem protocol API.
-func NewPublicEthereumAPI(b Backend) *PublicEthereumAPI {
-	return &PublicEthereumAPI{b}
+func NewPublicEthereumAPI(b Backend, solcPath *string, solc **compiler.Solidity) *PublicEthereumAPI {
+	return &PublicEthereumAPI{b, solcPath, solc}
```

Comment:
I see why you did it like this now. The issue is that the admin API wants to update the path and the ethereum API wants to use it. I will clean this up later, too, so the *compiler.Solidity handles this internally by providing SetPath.